src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -60,9 +58,42 @@ class TosService(val directoryDao: DirectoryDAO, val tosConfig: TermsOfServiceCo
   private def isRollingWindowInEffect() = tosConfig.rollingAcceptanceWindowExpiration.exists(Instant.now().isBefore(_))
 
   @Deprecated
-  def getTosDetails(samUser: SamUser, samRequestContext: SamRequestContext): IO[TermsOfServiceDetails] =
+  def getTosDetails(samUser: SamUser, samRequestContext: SamRequestContext): IO[OldTermsOfServiceDetails] =
     directoryDao.getUserTos(samUser.id, samRequestContext).map { tos =>
-      TermsOfServiceDetails(isEnabled = true, tosConfig.isGracePeriodEnabled, tosConfig.version, tos.map(_.version))
+      OldTermsOfServiceDetails(isEnabled = true, tosConfig.isGracePeriodEnabled, tosConfig.version, tos.map(_.version))
+    }
+
+  def getTermsOfServiceDetailsForUser(
+      requestedUserId: WorkbenchUserId,
+      requestingUser: SamUser,
+      isAdmin: Boolean,
```

Comment:
I don't think `isAdmin` should be passed to this function. If it is, then `isWorkbenchAdmin` needs to be run for every single invocation of this method, which is way more than is necessary.

Reply:
I think in general, if we want to deprecate "admin" routes, we would need to know if the caller is an admin to do our business logic on it. And I do not think it should be done inside the routes layers since I think that should be pretty much absent from logic.

Reply:
We need to then push the check for who's an admin or not down to the service layer. We can't have every single call to an endpoint check to see if the user is an admin.